tests/TESTS/common/device.py

```diff
@@ -77,6 +86,16 @@ def wait_for_output(self, search: str, timeout: float = 10, assert_timeout: bool
         :return: Line received before a match
         :rtype: list of str
         """
+        loop = asyncio.get_running_loop()
+        async with self._wait_for_output_lock:
+            try:
+                return await loop.run_in_executor(
+                    self._wait_for_output_executor, self._wait_for_output, search, timeout, assert_timeout)
+            except asyncio.CancelledError:
+                await asyncio.shield(self._cancel_wait_for_output)
```

Comment:
what causes this?

Reply:
[loop.run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) if the [Future](https://docs.python.org/3/library/asyncio-future.html) is cancelled. Currently, the shielded awaitable is a NOP.

Reply:
how is the future cancelled?

Reply:
It could be cancelled by the user.